src/models.py

```diff
@@ -0,0 +1,178 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+class Deshadower(nn.Module):
+    def __init__(self, in_channels, out_channels, res_blocks=9):
+        super(Deshadower, self).__init__()
+
+        # Initial conv layer
+        self.model = nn.Sequential(
+            nn.ReflectionPad2d(3),
+            nn.Conv2d(in_channels, 64, 7),
+            nn.InstanceNorm2d(64),
+            nn.ReLU(inplace=True),
+        )
+
+        # downsampling
+        in_channels = 64
+        out_channels = in_channels * 2
+
+        downsampling = (
+            nn.Conv2d(in_channels, out_channels, kernel_size=3, stride=2, padding=1),
+            nn.InstanceNorm2d(out_channels),
+            nn.ReLU(inplace=True),
+        )
+
+        for _ in range(2):
+            self.model, *_ = map(self.model.append, downsampling)
+            in_channels = out_channels
+            out_channels = in_channels * 2
+
+        # residual blocks
+        residual_block = (
+            nn.ReflectionPad2d(1),
+            nn.Conv2d(in_channels, in_channels, kernel_size=3),
+            nn.InstanceNorm2d(in_channels),
+            nn.ReLU(inplace=True),
+            nn.ReflectionPad2d(1),
+            nn.Conv2d(in_channels, in_channels, 3),
+            nn.InstanceNorm2d(in_channels),
+        )
+
+        if res_blocks <= 3:
+            res_blocks = 9
+
+        for _ in range(res_blocks):
+            self.model, *_ = map(self.model.append, residual_block)
+
+        # upsampling
+        out_channels = in_channels // 2
+        upsampling = (
+            nn.ConvTranspose2d(
+                in_channels, out_channels, 3, stride=2, padding=1, output_padding=1
+            ),
+            nn.InstanceNorm2d(out_channels),
+            nn.ReLU(inplace=True),
+        )
+
+        for _ in range(2):
+            self.model, *_ = map(self.model.append, upsampling)
+            in_channels = out_channels
+            out_channels = in_channels // 2
+
+        # output layer
+        self.model.append(nn.ReflectionPad2d(3))
+        self.model.append(nn.Conv2d(64, out_channels, 7))
+
+    def forward(self, x):
+        return (self.model(x) + x).tanh()
+
+
+class Shadower(nn.Module):
+    def __init__(self, in_channels, out_channels, res_blocks=9):
+        super(Shadower, self).__init__()
+
+        # initial conv layer
+        self.model = nn.Sequential(
+            nn.ReflectionPad2d(3),
+            nn.Conv2d(in_channels + 1, 64, 7),  # + mask
+            nn.InstanceNorm2d(64),
+            nn.ReLU(inplace=True),
+        )
+
+        # downsampling
+        in_channels = 64
```

Comment:
overwritting

Reply:
Same issue as before, after testing we can alter the overwritting.

Reply:
done